core/execution.go

```diff
@@ -85,6 +85,11 @@ func exec(env vm.Environment, caller vm.ContractRef, address, codeAddr *common.A
 		createAccount = true
 	}
 
+	// mark the code hash if the execution is a call, callcode or delegate.
```

Comment:
This comment is not very helpful if you don't know the background why some code needs to be marked as special. I would suggest removing this comment line and add a bit more explanation to the docs for `Environment.MarkCodeHash` why such a feature is available and used for. In `execDelegateCall` there is no comment for the same logic.